test/library/encryption/assertion.go

```diff
@@ -56,6 +72,42 @@ func AssertSecretOfLifeEncrypted(t testing.TB, clientSet ClientSet, secretOfLife
 	}
 }
 
+func AssertEncryptionConfigForSecretsAndConfigMaps(t testing.TB, clientSet ClientSet) {
+	t.Helper()
+	t.Logf("Checking if %q in %q has desired GRs %v", "encryption-config-openshift-kube-apiserver", operatorclient.GlobalMachineSpecifiedConfigNamespace, defaultTargetGRs)
+	encryptionCofnigSecret, err := clientSet.Kube.CoreV1().Secrets(operatorclient.GlobalMachineSpecifiedConfigNamespace).Get("encryption-config-openshift-kube-apiserver", metav1.GetOptions{})
+	require.NoError(t, err)
+	encodedEncryptionConfig, foundEncryptionConfig := encryptionCofnigSecret.Data["encryption-config"]
+	if !foundEncryptionConfig {
+		t.Errorf("Haven't found encryption config at %q key in the encryption secret %q", "encryption-config", "encryption-config-openshift-kube-apiserver")
+	}
+
+	decoder := apiserverCodecs.UniversalDecoder(apiserverconfigv1.SchemeGroupVersion)
+	encryptionConfigObj, err := runtime.Decode(decoder, encodedEncryptionConfig)
+	require.NoError(t, err)
+	encryptionConfig, ok := encryptionConfigObj.(*apiserverconfigv1.EncryptionConfiguration)
+	if !ok {
+		t.Errorf("Unable to decode encryption config, unexpected wrong type %T", encryptionConfigObj)
+	}
+
+	for _, rawActualResource := range encryptionConfig.Resources {
+		if len(rawActualResource.Resources) != 1 {
+			t.Errorf("Invalid encryption config for resource %s, expected exactly one resource, got %d", rawActualResource.Resources, len(rawActualResource.Resources))
+		}
+		actualResource := schema.ParseGroupResource(rawActualResource.Resources[0])
+		actualResourceFound := false
+		for _, expectedResource := range defaultTargetGRs {
+			if reflect.DeepEqual(expectedResource, actualResource) {
+				actualResourceFound = true
+				break
+			}
+		}
+		if !actualResourceFound {
+			t.Errorf("Encryption config has an invalid resource %v", actualResource)
+		}
+	}
+}
+
```

Comment:
this only verifies the format. Shouldn't we check the contents? Compare my integration test where I check against this  summary string https://github.com/openshift/library-go/blob/master/test/e2e-encryption/encryption_test.go#L332.